crates/compiler/fmt/src/def.rs

```diff
@@ -529,14 +529,21 @@ impl<'a> Formattable for TypeHeader<'a> {
         _newlines: Newlines,
         indent: u16,
     ) {
+        let old_flags = buf.flags;
+        buf.flags = MigrationFlags {
+            parens_and_commas: false,
+            ..old_flags
+        };
```

Comment:
This looks somewhat suspicious. Why do we need to do this?

Reply:
Type header uses patterns, but Richard specifically said nothing about type annotations is to change

Reply:
So this never allows migration to parens_and_commas

Reply:
That's why the PatternApplyStyle is fixed to whitespace.  So if someone writes PNC style - which will be parsed - it will be formatted as WS style.

Reply:
Hmm. I find it interesting that `TypeHeader` has `vars` being a list of `Pattern`s - I wonder if that should be changed?

As is, this seems like a bit of a code smell.